From 6ea92aecb99bdb89b0c2180a7c522d520fed661d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:24 -0400
Subject: [usb] sierra: remove unimplemented set_termios

Message-id: <1410887463-20674-433-git-send-email-dzickus@redhat.com>
Patchwork-id: 95638
O-Subject: [RHEL7 PATCH 432/591] USB: sierra: remove unimplemented set_termios
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e825aaa0624ef5cc0aa06e3102ddfc9ed95c2e2c
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:01 2014 +0200

    USB: sierra: remove unimplemented set_termios

    The driver does not implement set_termios so the operation can be left
    unset (tty will do the tty_termios_copy_hw for us).

    Note that the send_setup call is bogus as it really only sets DTR/RTS
    to their current values.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index fa0b78a..854ac61 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -372,13 +372,6 @@ static int sierra_send_setup(struct usb_serial_port *port)
  return retval;
 }
 
-static void sierra_set_termios(struct tty_struct *tty,
-  struct usb_serial_port *port, struct ktermios *old_termios)
-{
- tty_termios_copy_hw(&tty->termios, old_termios);
- sierra_send_setup(port);
-}
-
 static int sierra_tiocmget(struct tty_struct *tty)
 {
  struct usb_serial_port *port = tty->driver_data;
@@ -1079,7 +1072,6 @@ static struct usb_serial_driver sierra_device = {
  .write             = sierra_write,
  .write_room        = sierra_write_room,
  .chars_in_buffer   = sierra_chars_in_buffer,
- .set_termios       = sierra_set_termios,
  .tiocmget          = sierra_tiocmget,
  .tiocmset          = sierra_tiocmset,
  .attach            = sierra_startup,
-- 
1.7.1